assets/src/components/detours/detourMap.tsx

```diff
@@ -1,34 +1,57 @@
-import React, { useState } from "react"
-import { Shape } from "../../schedule"
-import { LatLngExpression } from "leaflet"
+import React, { useEffect, useState } from "react"
+import { Shape, ShapePoint } from "../../schedule"
+import { LatLngLiteral } from "leaflet"
 import { Polyline, useMapEvent } from "react-leaflet"
 import Leaflet from "leaflet"
 import Map from "../map"
 import { CustomControl } from "../map/controls/customControl"
 import { Button } from "react-bootstrap"
 import { ReactMarker } from "../map/utilities/reactMarker"
 import { closestPosition } from "../../util/math"
+import { fetchDetourDirections } from "../../api"
 
 export const DetourMap = ({ shape }: { shape: Shape }) => {
-  const [startPoint, setStartPoint] = useState<LatLngExpression | null>(null)
-  const [endPoint, setEndPoint] = useState<LatLngExpression | null>(null)
-  const [detourPositions, setDetourPositions] = useState<LatLngExpression[]>([])
+  const [startPoint, setStartPoint] = useState<LatLngLiteral | null>(null)
+  const [endPoint, setEndPoint] = useState<LatLngLiteral | null>(null)
+  const [waypoints, setWaypoints] = useState<LatLngLiteral[]>([])
+  const [detourShapePositions, setDetourShapePositions] = useState<
+    LatLngLiteral[]
+  >([])
 
-  const onAddDetourPosition = (p: LatLngExpression) =>
-    setDetourPositions((positions) => [...positions, p])
+  const onAddDetourPosition = (p: LatLngLiteral) => {
+    setWaypoints((positions) => [...positions, p])
+  }
+
+  useEffect(() => {
+    const shapePoints: ShapePoint[] = waypoints.map(
+      (waypoint: LatLngLiteral) => {
+        const { lat, lng } = waypoint
+        return { lat, lon: lng }
+      }
+    )
```

Comment:
**suggestion:** since this operation and its inverse happen in a few places it might be worth splitting it out into utility functions, even if it's small / simple logic.